citydb-database/src/main/java/org/citydb/database/metadata/SpatialReference.java

```diff
@@ -59,4 +68,45 @@ public String getURI() {
     public String getWKT() {
         return wkt;
     }
+
+    public Optional<CoordinateReferenceSystem> getDefinition() {
+        return Optional.ofNullable(definition.get());
+    }
+
+    public int getDimension() {
+        if (definition.get() != null) {
+            return definition.get().getCoordinateSystem().getDimension();
+        } else {
+            return type == SpatialReferenceType.COMPOUND_CRS
+                    || type == SpatialReferenceType.GEOGRAPHIC3D_CRS ?
+                    3 : 2;
+        }
+    }
+
+    private CoordinateReferenceSystem buildDefinition() {
+        CoordinateReferenceSystem crs = null;
+        try {
+            crs = CRS.decode("EPSG:" + srid);
+        } catch (Exception e) {
+            //
+        }
+
+        if (crs == null) {
+            try {
+                crs = CRS.decode(uri);
+            } catch (Exception e) {
+                //
+            }
+        }
+
+        if (crs == null) {
+            try {
+                crs = CRS.parseWKT(wkt);
```

Comment:
Should the wkt take precedence than srid and crs? because, the wkt can be customized and defined by users for specific use cases e.g. (swap X/Y axis).

Reply:
done with 58489a9